### PR TITLE
Deprecate fetch_url, use requests instead

### PR DESCRIPTION
**Pull request by happz**
The vast majority of the patch is just extending logging subsystem to
cover known ways requests logs its actions - its logger, httplib
internal logging - and how to plug both sides together.

### CODE REVIEW

File: gluetool/log.py [hunk 1]
```diff
@@ -515,6 +584,54 @@ def get_logger():
...
+    @staticmethod
+    def enable_logger_sentry(logger):
+        if not Logging.sentry:
+            return
+
+        Logging.sentry.enable_logging_breadcrumbs(logger)
+
+    OUR_LOGGERS = (
+        logging.getLogger('gluetool'),
+        logging.getLogger("urllib3")
```
Comment (thrix) [inline gluetool/log.py]: nitpicking, maybe use also single quotes here, looks a bit weird while reading

**Comment by thrix** [comment]
Hmm, this looks nice, I guess we could have a deep dive on the concepts here on one of our development meetings ;)

**Comment by happz** [comment]
@thrix Sure, I can show a demo. Although this could be even nicer if we'd have time to implement that extra verbose log file.

File: gluetool/log.py (hunk 1 above)
Comment (happz) [reply gluetool/log.py]: Right, fixed.

**Review by mkluson — APPROVED** [review]

**Review by thrix — APPROVED** [review]
Tested manually and it works like charm :)

Very very very nice feature, love it

```
[09:39:24] [+] [greenwave] list of policies: https://some-url.com/api/v1.0/policies
[09:39:24] [D] waiting for condition 'trying to get response from Greenwave', timeout 299 seconds, check every 5 seconds
[09:39:24] [D] calling callback function
[09:39:24] [D] Starting new HTTPS connection (1): some-url.com
[09:39:25] [D] https://some-url.com:443 "POST /api/v1.0/decision HTTP/1.1" 200 5532
[09:39:25] [D] check passed, assuming success
[09:39:25] [D] opening URL 'https://some-url.com/api/v1.0/policies'
```

**Review by optak — APPROVED** [review]
LGTM

**Pull request merged**